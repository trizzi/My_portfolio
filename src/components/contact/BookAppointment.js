import React from 'react';
import { InlineWidget } from 'react-calendly';

const BookAppointment = () => {
	return (
		<div
			className='bg-black h-[100vh]'
			data-testid='calendly-widget'>
			<InlineWidget url='https://calendly.com/patrickakhamiogu' />
		</div>
	);
};

export default BookAppointment;
