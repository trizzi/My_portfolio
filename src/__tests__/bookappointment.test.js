import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeContextProvider } from '../context/ThemeContext';

import BookAppointment from '../components/contact/BookAppointment';

test('renders BookAppointment component with Calendly inline widget', () => {
	// Render the BookAppointment component
	const { getByTestId } = render(
		<ThemeContextProvider>
			<BookAppointment />
		</ThemeContextProvider>,
	);

	// Find the Calendly inline widget by its test ID
	const calendlyWidget = getByTestId('calendly-widget');

	// Assert that the Calendly inline widget is present
	expect(calendlyWidget).toBeInTheDocument();

	expect(getByTestId).toMatchSnapshot();
});
