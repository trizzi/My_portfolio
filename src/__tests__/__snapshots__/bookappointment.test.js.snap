// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BookAppointment component with Calendly inline widget 1`] = `[Function]`;
